### PR TITLE
E2E Tests for GameServer Player Tracking

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

Updated the udp-simple example to accept various player tracking
commands, and also implemented e2e tests to test it working on a
singular GameServer instance.

**Which issue(s) this PR fixes**:

Work on #1507

**Special notes for your reviewer**:

Once people have a chance to review the udp-simple code, I'll make and push the image, so on first run, the build is going to fail.

### CODE REVIEW

File: test/e2e/gameserver_test.go [hunk 1]
```diff
@@ -736,3 +736,161 @@ func TestGameServerResourceValidation(t *testing.T) {
...
+
+		reply, err = e2eframework.SendGameServerUDP(gs, "PLAYER_CAPACITY 20")
+		if err != nil {
+			t.Fatalf("Could not message GameServer: %v", err)
+		}
+		assert.Equal(t, "ACK: PLAYER_CAPACITY 20\n", reply)
+
+		reply, err = e2eframework.SendGameServerUDP(gs, "PLAYER_CAPACITY")
+		assert.NoError(t, err)
+		assert.Equal(t, "20", reply)
```
Comment (aLekSer) [inline test/e2e/gameserver_test.go:767]: I think new line is needed here and in all other responses.
```suggestion
		assert.Equal(t, "20\n", reply)
```

File: examples/simple-udp/main.go [hunk 2]
```diff
@@ -176,6 +176,52 @@ func readWriteLoop(conn net.PacketConn, stop chan struct{}, s *sdk.SDK) {
...
+		case "PLAYER_CONNECTED":
+			if len(parts) < 2 {
+				respond(conn, sender, "ERROR: Invalid PLAYER_CONNECTED, should have 1 arguments\n")
+				continue
+			}
+			respond(conn, sender, playerIsConnected(s, parts[1]))
+			continue
+
+		case "GET_PLAYERS":
+			respond(conn, sender, getConnectedPlayers(s))
```
Comment (aLekSer) [inline examples/simple-udp/main.go:219]: There should be new line symbol to comply with other responses and for better output in `nc -u`. Something like:
```suggestion
			respond(conn, sender, fmt.Sprintf("%s\n", getConnectedPlayers(s)))
```

File: examples/simple-udp/main.go [hunk 3]
```diff
@@ -176,6 +176,52 @@ func readWriteLoop(conn net.PacketConn, stop chan struct{}, s *sdk.SDK) {
...
+			continue
+
+		case "PLAYER_COUNT":
+			respond(conn, sender, getPlayerCount(s))
```
Comment (aLekSer) [inline examples/simple-udp/main.go:223]: Same as above, new line symbol.

**Comment by aLekSer** [comment]
Overall is great. But it seems that `Makefile` also should be updated https://github.com/googleforgames/agones/blob/master/build/Makefile#L60

**Comment by markmandel** [comment]
Review updates in place, if you are happy with me pushing up the image, I will do so.

**Comment by markmandel** [comment]
Image pushed!

**Comment by aLekSer** [comment]
Build was successful, please merge when you think it is suitable.

**Pull request merged**